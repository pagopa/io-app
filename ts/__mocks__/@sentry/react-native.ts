export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
