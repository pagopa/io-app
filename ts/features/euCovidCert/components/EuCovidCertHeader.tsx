import * as React from "react";
import { View, Image, StyleSheet } from "react-native";
import { WithEUCovidCertificateHeaderData } from "../types/EUCovidCertificate";
import { H1 } from "../../../components/core/typography/H1";
import { H2 } from "../../../components/core/typography/H2";
import { isStringNullyOrEmpty } from "../../../utils/strings";

const styles = StyleSheet.create({
  row: {
    flexDirection: "row",
    justifyContent: "space-between"
  },
  column: { flexDirection: "column", flex: 1 },
  logo: {
    width: 84,
    height: 84
  }
});

/**
 * header with
 * | title       | logo |
 * | subtitle    |      |
 * @param props
 * @constructor
 */
export const EuCovidCertHeader = (props: WithEUCovidCertificateHeaderData) => (
  <>
    <View style={styles.row}>
      <View style={styles.column}>
        <H1 testID={"EuCovidCertHeaderTitle"}>{props.headerData.title}</H1>
        <H2 testID={"EuCovidCertHeaderSubTitle"}>
          {props.headerData.subTitle}
        </H2>
      </View>
      {/* it could happen we don't want to show any logo, in that case url will be empty */}
      {!isStringNullyOrEmpty(props.headerData.logoUrl) && (
        <Image
          testID={"EuCovidCertHeaderLogoID"}
          source={{ uri: props.headerData.logoUrl }}
          style={styles.logo}
          importantForAccessibility={"no"}
          accessibilityElementsHidden={true}
          resizeMode={"contain"}
        />
      )}
    </View>
  </>
);
