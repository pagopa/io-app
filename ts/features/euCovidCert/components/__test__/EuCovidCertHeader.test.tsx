import { render } from "@testing-library/react-native";
import React from "react";
import { EuCovidCertHeader } from "../EuCovidCertHeader";

const headerData = {
  title: "title",
  subTitle: "subtitle",
  logoUrl: "logoUrl"
};
describe("EuCovidCertHeader", () => {
  it(`it should match the snapshot`, () => {
    const component = render(<EuCovidCertHeader headerData={headerData} />);
    expect(component).toMatchSnapshot();
  });
});

describe("when the headerData is full filled", () => {
  it(`it should show the title, subtitle and logo`, () => {
    const component = render(<EuCovidCertHeader headerData={headerData} />);
    expect(component.queryByText(headerData.title)).not.toBeNull();
    expect(component.queryByText(headerData.subTitle)).not.toBeNull();
    expect(component.queryByTestId("EuCovidCertHeaderLogoID")).not.toBeNull();
  });
});

describe("when title and subtitle are empty", () => {
  it(`it should show the title, subtitle and the logo`, () => {
    const component = render(
      <EuCovidCertHeader
        headerData={{ title: "", subTitle: "", logoUrl: "id" }}
      />
    );
    expect(component.queryByTestId("EuCovidCertHeaderTitle")).not.toBeNull();
    expect(component.queryByTestId("EuCovidCertHeaderSubTitle")).not.toBeNull();
    expect(component.queryByTestId("EuCovidCertHeaderLogoID")).not.toBeNull();
  });
});

describe("when the logo url is empty", () => {
  it(`it should show the title, subtitle and not the logo`, () => {
    const component = render(
      <EuCovidCertHeader headerData={{ ...headerData, logoUrl: "" }} />
    );
    expect(component.queryByText(headerData.title)).not.toBeNull();
    expect(component.queryByText(headerData.subTitle)).not.toBeNull();
    expect(component.queryByTestId("EuCovidCertHeaderLogoID")).toBeNull();
  });
});

describe("when the logo url is filled with blank spaces", () => {
  it(`it should show the title, subtitle and not the logo`, () => {
    const component = render(
      <EuCovidCertHeader headerData={{ ...headerData, logoUrl: "    " }} />
    );
    expect(component.queryByText(headerData.title)).not.toBeNull();
    expect(component.queryByText(headerData.subTitle)).not.toBeNull();
    expect(component.queryByTestId("EuCovidCertHeaderLogoID")).toBeNull();
  });
});
