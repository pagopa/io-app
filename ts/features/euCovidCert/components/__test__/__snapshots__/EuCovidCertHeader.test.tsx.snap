// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuCovidCertHeader it should match the snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="Bold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 26,
          "lineHeight": 32,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 26,
            "lineHeight": 32,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "700",
          },
        ]
      }
      testID="EuCovidCertHeaderTitle"
      weight="Bold"
    >
      title
    </Text>
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="Bold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 20,
          "lineHeight": 24,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 20,
            "lineHeight": 24,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "700",
          },
        ]
      }
      testID="EuCovidCertHeaderSubTitle"
      weight="Bold"
    >
      subtitle
    </Text>
  </View>
  <Image
    accessibilityElementsHidden={true}
    importantForAccessibility="no"
    resizeMode="contain"
    source={
      Object {
        "uri": "logoUrl",
      }
    }
    style={
      Object {
        "height": 84,
        "width": 84,
      }
    }
    testID="EuCovidCertHeaderLogoID"
  />
</View>
`;
